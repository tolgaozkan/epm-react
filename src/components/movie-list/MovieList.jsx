import React from 'react';

require('./style.scss');

export default class MovieList extends React.Component {
  render() {
    let movies;

    if (this.props && this.props.movies) {
      console.log("this.props", this.props);
      movies = this.props.movies.map((movie, i) => (
        <div className="movie" key={i}>
            <p className="movie-name">Name: {movie.title}</p>
            <p className="movie-genre">Genre: {movie.genres.join(', ')}</p>
          </div>
      ));
    }
    return (
      <div className="movies-container">{movies}</div>
    );
  }
}

