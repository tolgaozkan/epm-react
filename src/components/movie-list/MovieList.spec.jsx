import * as React from 'react';
import { create as render } from 'react-test-renderer';

import MovieList from './MovieList';

it('renders movie list', () => {
  const result = render(<MovieList />);
  const snap = result.toJSON();

  expect(snap).toMatchSnapshot();
});
