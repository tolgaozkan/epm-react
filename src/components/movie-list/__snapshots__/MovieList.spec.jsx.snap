// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders movie list 1`] = `
<div
  className="movies-container"
/>
`;
